### DISCUSSION TITLE
How to toggle menu on hover?

### DISCUSSION THREAD

**Discussion by amirhhashemi** [adobe/react-spectrum#3482] [Q&A] [answered]
I'm trying to create a navigation menu like this:

https://www.radix-ui.com/docs/primitives/components/navigation-menu

The reason that I'm not using Radix UI is that I'm doing this in Solid.js not React. Solid.js has a library named `solid-aria` which is the equivalent of react-aria in Solid but there is no such library for Radix UI.

Currently, the menu can be toggled only with the "press" and "longPress" events. Is there a way to use "hover" for that?

**Comment by snowystinger** [comment, marked as answer]
You can control the state using the stately package. And you can use useHover to listen for hover changes. Then you can open whenever you get a hover change true.
A word of caution. Make sure you also open for press and long press for accessibility. Do not rely on hover. Otherwise you'll find yourself broken on mobile.

**Reply by amirhhashemi** [reply]
Hey @snowystinger. Thanks for the reply. I ended up combining `useHover` with `useButton` to achieve this and it works. Thank you.
